rfcs/text/0352-graph-type-system.md

```diff
@@ -0,0 +1,2579 @@
+- Feature Name: graph-type-system
+- Start Date: 2022-06-14
+- RFC PR: [blockprotocol/blockprotocol#352](https://github.com/blockprotocol/blockprotocol/pull/352)
+- Block Protocol Discussion: [blockprotocol/blockprotocol#0000](https://github.com/blockprotocol/blockprotocol/discussions/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC proposes changing the way data structural requirements are expressed within the BP.
+
+Rather than allowing any free-form JSON schema, this RFC defines a set of JSON schema meta-schemas, which describe:
```

Comment:
Cool, have made the change in https://github.com/blockprotocol/blockprotocol/pull/352/commits/736eeb99639134af39bd5fe12dc4f74fc32f3d97

Have slightly tweaked the added sentence further:
> The structure of these types are defined by (and can be validated against) their meta-schemas, which can be found in the [reference-level explanation](#reference-level-explanation).

My thinking behind referring to them as "meta-schemas" rather than "JSON schema meta-schemas" is because I find its too similar to "JSON schema meta-schema", which I would interpret as a reference to the meta-schema of JSON schema (draft-07, draft-05, etc). This also feels like it better matches how "meta-schemas" are referred to in the [JSON-Schema spec](https://json-schema.org/specification.html).

Reply:
That's fair enough, I'm fine with the change, although I think meta-schema on it's own is pretty vague and we're specifically talking about a JSON Schema which _is_ a meta-schema (i.e. it validates other JSON schema)

Anyway changes are good